### DISCUSSION TITLE
[FR] free DDNS service

### DISCUSSION THREAD

**Discussion by Macka323** [pivpn/pivpn#1401] [Feature requests]
**Is your feature request related to a problem? Please describe.**
It would be nice to have built in ddns services like:
no-ip,
duckdns, (recommended) because they are free
now-dns, (also recommended) because they are free
etc.

**Describe the solution you'd like**
To have built in option to connect ddns service

ex. comand 
pivpn -d (to add ddns service)
 // list the available options
// and chose from them 
// then put your key or username/password 
// chose update rate in minutes or seconds

**Describe alternatives you've considered**
//

**Additional context**
i think that their APIs are simple to implement

**Comment by coolapso** [comment]
Moved to discussions in light of recent Updates on documentation: https://github.com/pivpn/pivpn#feedback--support 

Thanks for your feature request. 

Even tough i understand why this is being requested, I don't think this should be in the scope of PiVPN, Maximum we could point towards ddns services documentation similar fashion as we already do here: https://docs.pivpn.io/guides/dynamic-dns/ 

Why? 

PiVPN goal is one and simple, to install a vpn protocol with reasonably hardened settings by default! and i've learned a lesson in the past, when we tried to install bitwarden with pivpn. 

Adding more stuff for PiVPN to install, means adding more dependencies and instead of a VPN installer it becomes a many things installer, which can work differently in so many other systems, this brings a lot of overhead work and becomes extremely painful to maintain as we have to keep track of every single piece of software we install, and then become a support hub for every single one of it.


I am a very big advocate of simple things that are very good at doing one thing, while still being able to integrate with other projects. 

This case there is not really big integration, however helping out with documentation is always a good way to help out!
